src/ftpclass.cc

```diff
@@ -1535,7 +1535,7 @@ int   Ftp::Do()
 	    }
 	 }
 #if USE_SSL
-	 if(strcmp(proxy_auth_type,"joined") || strcmp(proxy_auth_type,"joined-acct"))
+	 if(strcmp(proxy_auth_type,"joined") != 0 && strcmp(proxy_auth_type,"joined-acct") != 0)
```

Comment:
I think it would be nicer to have a "bool use_auth=false" before proxy type checks and set it to true in the "open" and "proxy-user@host" cases, then check the variable and send AUTH if true.
Also it would be nice to avoid the ugly (user && pass) check in two places.